File: app/src/main/java/im/tox/antox/ChatMessages.java
package im.tox.antox;

public class ChatMessages {
    public String message;
    public String time;
    public boolean ownMessage;
    public String friendName;

    public ChatMessages() {
        super();
    }

    public ChatMessages(String message, String time, boolean ownMessage) {
        super();
        this.time = time;
        this.message = message;
        this.ownMessage = ownMessage;
    }

    public ChatMessages(String message, String time, String friendName) {
        super();
        this.time = time;
        this.message = message;
        this.friendName = friendName;
    }

    public boolean IsMine() {
        return ownMessage;
    }
}

File: app/src/main/scala/im/tox/antox/wrapper/ChatMessages.scala
package im.tox.antox.wrapper

import java.sql.Timestamp

import im.tox.antox.wrapper.MessageType.MessageType

class ChatMessages(
  val id: Int,
  val message_id: Int,
  val key: String,
  val sender_name: String,
  val message: String,
  val time: Timestamp,
  val received: Boolean,
  val sent: Boolean,
  val size: Int,
  val `type`: MessageType,
  val fileKind: FileKind) {

  def isMine: Boolean = {
    `type` == MessageType.OWN || `type` == MessageType.FILE_TRANSFER || `type` == MessageType.GROUP_OWN
  }

  def getType: MessageType = `type`

  override def toString: String = message
}
